### ISSUE TITLE
Fail to redirect to the admin page

### ISSUE THREAD

**Issue by Spy-Seth** [EventStore/EventStore#990]
If you try to access directly to a stream url in the browser, EventStore failed to redirect you to the admin UI for this stream. It return an HTML page that should work but it return it with the `Content-Type:text/plain;`. The browser never execute the HTML/JS inside the page and just print it as text.

Example uri: `http://evenstore.localtest:2113/streams/a-stream-name-that-exist/0`

Reponse content:
```html
<!DOCTYPE html>
<html>
<head>
</head>
<body>
<script>
    var data = {};
    var newLocation = '/web/index.html#/streams/' + data.streamId
    if('positionEventNumber' in data){
        newLocation = newLocation + '/' + data.positionEventNumber;
    }
    window.location.replace(newLocation);
</script>
</body>
</html>
```

Headers:
```
HTTP/1.1 404 Not Found
Access-Control-Allow-Methods: GET, OPTIONS
Access-Control-Allow-Headers: Content-Type, X-Requested-With, X-Forwarded-Host, X-PINGOTHER, Authorization, ES-LongPoll, ES-ExpectedVersion, ES-EventId, ES-EventType, ES-RequiresMaster, ES-HardDelete, ES-ResolveLinkTo
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: Location, ES-Position, ES-CurrentVersion
Content-Type: text/plain; charset=utf-8
Server: Mono-HTTPAPI/1.0
Date: Fri, 29 Jul 2016 13:17:44 GMT
Content-Length: 513
Keep-Alive: timeout=15,max=100
```

**Comment by gregoryyoung** [comment]
Thats not a stream uri that is an event uri (/0)

Stream does:
 ✗ curl -v http://127.0.0.1:2113/streams/test-stream -H "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 2113 (#0)
> GET /streams/test-stream HTTP/1.1
> Host: 127.0.0.1:2113
> User-Agent: curl/7.43.0
> Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
>
< HTTP/1.1 200 OK
< Access-Control-Allow-Methods: POST, DELETE, GET, OPTIONS
< Access-Control-Allow-Headers: Content-Type, X-Requested-With, X-Forwarded-Host, X-PINGOTHER, Authorization, ES-LongPoll, ES-ExpectedVersion, ES-EventId, ES-EventType, ES-RequiresMaster, ES-HardDelete, ES-ResolveLinkTo
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: Location, ES-Position, ES-CurrentVersion
< Cache-Control: max-age=0, no-cache, must-revalidate
< Vary: Accept
< ETag: "0;1268450035"
< Content-Type: text/html; charset=utf-8
< Server: Mono-HTTPAPI/1.0
< Date: Fri, 29 Jul 2016 14:47:11 GMT
< Content-Length: 2114
< Keep-Alive: timeout=15,max=100
<

            <!DOCTYPE html>
            <html>
            <head>
            </head>
            <body>
            <script>
                var data = {
  "title": "Event stream 'test-stream'",
  "id": "http://127.0.0.1:2113/streams/test-stream",
  "updated": "2016-07-25T16:42:35.080036Z",
  "streamId": "test-stream",
  "author": {
    "name": "EventStore"
  },
  "headOfStream": true,
  "selfUrl": "http://127.0.0.1:2113/streams/test-stream",
  "eTag": "0;-2060438500",
  "links": [
    {
      "uri": "http://127.0.0.1:2113/streams/test-stream",
      "relation": "self"
    },
    {
      "uri": "http://127.0.0.1:2113/streams/test-stream/head/backward/20",
      "relation": "first"
    },
    {
      "uri": "http://127.0.0.1:2113/streams/test-stream/1/forward/20",
      "relation": "previous"
    },
    {
      "uri": "http://127.0.0.1:2113/streams/test-stream/metadata",
      "relation": "metadata"
    }
  ],
  "entries": [
    {
      "eventId": "850fa1eb-0d39-4c19-9337-16d0d74b8171",
      "eventType": "TakeSomeSpaceEvent",
      "eventNumber": 0,
      "streamId": "test-stream",
      "isJson": false,
      "isMetaData": false,
      "isLinkMetaData": false,
      "positionEventNumber": 0,
      "positionStreamId": "test-stream",
      "title": "0@test-stream",
      "id": "http://127.0.0.1:2113/streams/test-stream/0",
      "updated": "2016-07-25T16:42:35.080036Z",
      "author": {
        "name": "EventStore"
      },
      "summary": "TakeSomeSpaceEvent",
      "links": [
        {
          "uri": "http://127.0.0.1:2113/streams/test-stream/0",
          "relation": "edit"
        },
        {
          "uri": "http://127.0.0.1:2113/streams/test-stream/0",
          "relation": "alternate"
        }
      ]
    }
  ]
};
                var newLocation = '/web/index.html#/streams/' + data.streamId
                if('positionEventNumber' in data){
                    newLocation = newLocation + '/' + data.positionEventNumber;
                }
                window.location.replace(newLocation);
            </script>
            </body>
            </html>
* Connection #0 to host 127.0.0.1 left intact
            %


I get correct for event uri as well

✗ curl -v http://127.0.0.1:2113/streams/test-stream/0 -H "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 2113 (#0)
> GET /streams/test-stream/0 HTTP/1.1
> Host: 127.0.0.1:2113
> User-Agent: curl/7.43.0
> Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
>
< HTTP/1.1 200 OK
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: Content-Type, X-Requested-With, X-Forwarded-Host, X-PINGOTHER, Authorization, ES-LongPoll, ES-ExpectedVersion, ES-EventId, ES-EventType, ES-RequiresMaster, ES-HardDelete, ES-ResolveLinkTo
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: Location, ES-Position, ES-CurrentVersion
< Cache-Control: max-age=31536000, public
< Vary: Accept
< Content-Type: text/html; charset=utf-8
< Server: Mono-HTTPAPI/1.0
< Date: Fri, 29 Jul 2016 14:48:30 GMT
< Content-Length: 1233
< Keep-Alive: timeout=15,max=100
<

            <!DOCTYPE html>
            <html>
            <head>
            </head>
            <body>
            <script>
                var data = {
  "eventId": "850fa1eb-0d39-4c19-9337-16d0d74b8171",
  "eventType": "TakeSomeSpaceEvent",
  "eventNumber": 0,
  "data": "test-data",
  "streamId": "test-stream",
  "isJson": false,
  "isMetaData": false,
  "isLinkMetaData": false,
  "positionEventNumber": 0,
  "positionStreamId": "test-stream",
  "title": "0@test-stream",
  "id": "http://127.0.0.1:2113/streams/test-stream/0",
  "updated": "2016-07-25T16:42:35.080036Z",
  "author": {
    "name": "EventStore"
  },
  "summary": "TakeSomeSpaceEvent",
  "links": [
    {
      "uri": "http://127.0.0.1:2113/streams/test-stream/0",
      "relation": "edit"
    },
    {
      "uri": "http://127.0.0.1:2113/streams/test-stream/0",
      "relation": "alternate"
    }
  ]
};
                var newLocation = '/web/index.html#/streams/' + data.streamId
                if('positionEventNumber' in data){
                    newLocation = newLocation + '/' + data.positionEventNumber;
                }
                window.location.replace(newLocation);
            </script>
            </body>
            </html>
* Connection #0 to host 127.0.0.1 left intact
            

Both also work in browser.


I do get your behaviour when I go to an event that does not exist:

curl -v http://127.0.0.1:2113/streams/test-stream/45 -H "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 2113 (#0)
> GET /streams/test-stream/45 HTTP/1.1
> Host: 127.0.0.1:2113
> User-Agent: curl/7.43.0
> Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
>
< HTTP/1.1 404 Not Found
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: Content-Type, X-Requested-With, X-Forwarded-Host, X-PINGOTHER, Authorization, ES-LongPoll, ES-ExpectedVersion, ES-EventId, ES-EventType, ES-RequiresMaster, ES-HardDelete, ES-ResolveLinkTo
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: Location, ES-Position, ES-CurrentVersion
< Content-Type: text/plain; charset=utf-8
< Server: Mono-HTTPAPI/1.0
< Date: Fri, 29 Jul 2016 14:49:35 GMT
< Content-Length: 513
< Keep-Alive: timeout=15,max=100
<

            <!DOCTYPE html>
            <html>
            <head>
            </head>
            <body>
            <script>
                var data = {};
                var newLocation = '/web/index.html#/streams/' + data.streamId
                if('positionEventNumber' in data){
                    newLocation = newLocation + '/' + data.positionEventNumber;
                }
                window.location.replace(newLocation);
            </script>
            </body>
            </html>
* Connection #0 to host 127.0.0.1 left intact
            %

**Comment by Spy-Seth** [comment]
I get this uri as a link in a subscription.

**Comment by gregoryyoung** [comment]
Its totally possible you could get it as a link in a subscription and it could be deleted by the time you come back depending on settings etc. That said if you look the last is identical to what you put no?

**Comment by Spy-Seth** [comment]
Not sure to understand the question. Could you rephrase please?

**Comment by pgermishuys** [comment]
The event you are attempting to access via that url is possibly not longer available. From the response that you are getting that appears to be the case. There could be a number of reasons for this which is what @gregoryyoung said.

The response from your request
```
HTTP/1.1 404 Not Found
Access-Control-Allow-Methods: GET, OPTIONS
Access-Control-Allow-Headers: Content-Type, X-Requested-With, X-Forwarded-Host, X-PINGOTHER, Authorization, ES-LongPoll, ES-ExpectedVersion, ES-EventId, ES-EventType, ES-RequiresMaster, ES-HardDelete, ES-ResolveLinkTo
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: Location, ES-Position, ES-CurrentVersion
Content-Type: text/plain; charset=utf-8
Server: Mono-HTTPAPI/1.0
Date: Fri, 29 Jul 2016 13:17:44 GMT
Content-Length: 513
Keep-Alive: timeout=15,max=100
```

**Comment by pgermishuys** [comment]
I am going to close this issue, please feel free to reopen it if you feel that the explanation provided is not helpful.
